pkg/lwip/contrib/sock/udp/lwip_sock_udp.c

```diff
@@ -120,23 +120,22 @@ ssize_t sock_udp_recv_buf(sock_udp_t *sock, void **data, void **ctx,
     if (remote != NULL) {
         /* convert remote */
         size_t addr_len;
-#if LWIP_IPV6
-        if (sock->base.conn->type & NETCONN_TYPE_IPV6) {
-            addr_len = sizeof(ipv6_addr_t);
-            remote->family = AF_INET6;
+        if (IS_USED(LWIP_IPV6)) {
+            if (!IS_USED(LWIP_IPV4) ||
+                (sock->base.conn->type & NETCONN_TYPE_IPV6))
+                {
+                addr_len = sizeof(ipv6_addr_t);
+                remote->family = AF_INET6;
+            }
         }
-        else {
-#endif
-#if LWIP_IPV4
+        else if (IS_USED(LWIP_IPV4)) {
             addr_len = sizeof(ipv4_addr_t);
             remote->family = AF_INET;
-#else
+        }
+        else {
             netbuf_delete(buf);
             return -EPROTO;
-#endif
-#if LWIP_IPV6
         }
-#endif
 #if LWIP_NETBUF_RECVINFO
         remote->netif = lwip_sock_bind_addr_to_netif(&buf->toaddr);
 #else
```

Comment:
Why is `IS_ACTIVE()` not used here?

Reply:
Unless I misread the documentation of lwIP, the `toaddr` member is only optionally available. Without `LWIP_NETBUF_RECVINFO` being set, this would access a non-existing struct member

Reply:
You are right. Had a look into the code and it is only available when `LWIP_NETBUF_RECVINFO` is 1.